### PR TITLE
Add process dump for failed integration tests

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?

This PR adds a dump of the processes running on the system when an integration test fails and saves it in a JSON file available as artifact

## Why is it important?

We are often encountering strange behavior in integration tests which can be attributed to extra processes running on the system and investigation benefits from having a picture of what was running on the system close to the time of failure.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Author's Checklist

- [ ]

## Related issues

- Relates #4164 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: pkg/testing/fixture_install.go [hunk 1]
```diff
@@ -170,6 +170,38 @@ func (f *Fixture) Install(ctx context.Context, installOpts *InstallOpts, opts ..
...
+			dir, err := findProjectRoot(f.caller)
+			if err != nil {
+				f.t.Logf("failed to dump process; failed to find project root: %s", err)
+				return
+			}
+
+			// Sub-test names are separated by "/" characters which are not valid filenames on Linux.
+			sanitizedTestName := strings.ReplaceAll(f.t.Name(), "/", "-")
+
+			filePath := filepath.Join(dir, "build", "diagnostics", fmt.Sprintf("TEST-%s-ProcessDump.json", sanitizedTestName))
```
Comment (rdner) [inline pkg/testing/fixture_install.go:185]: Did you manage to see a build where this artifact was attached? Can you link it here?

Comment (pchila) [reply pkg/testing/fixture_install.go:185]: It's in `build/diagnostics` , like in this failed run https://buildkite.com/elastic/elastic-agent/builds/7359#018dccc8-2c82-4845-9aa7-6440420fb850

Comment (rdner) [reply pkg/testing/fixture_install.go:185]: Great! Thanks!

**Review by AndersonQ — APPROVED** [review]

File: pkg/testing/fixture_install.go (hunk 1 above)
Comment (cmacknz) [reply pkg/testing/fixture_install.go:185]: We probably want to put the test architecture in the artifact name. You'll be able to tell windows from linux by the contents, but not arm vs x86 linux.

Diagnostics have the same naming problem, but the variables file tells you the host arch when you open them.

Comment (pchila) [reply pkg/testing/fixture_install.go]: added os and arch in 1f9348a8147471c5397c6c58725d71299b58f224

**Pull request merged**